sdk/identity/azure-identity/src/private/tenant_id_resolver.hpp

```diff
@@ -25,5 +25,7 @@ namespace Azure { namespace Identity { namespace _detail {
 
     // ADFS is the Active Directory Federation Service, a tenant ID that is used in Azure Stack.
     static bool IsAdfs(std::string const& tenantId);
+
+    static bool IsValidTenantId(std::string const& tenantId);
```

Comment:
Addresses the feedback to reduce duplication for this helper: https://github.com/Azure/azure-sdk-for-cpp/pull/5789#discussion_r1676517226